subprojects/files/src/main/java/org/gradle/api/internal/file/collections/ReproducibleDirectoryWalker.java

```diff
@@ -17,22 +17,71 @@
 package org.gradle.api.internal.file.collections;
 
 import org.apache.commons.io.comparator.PathFileComparator;
+import org.gradle.api.GradleException;
+import org.gradle.api.file.FileTreeElement;
+import org.gradle.api.file.FileVisitDetails;
+import org.gradle.api.file.FileVisitor;
+import org.gradle.api.file.RelativePath;
+import org.gradle.api.internal.file.DefaultFileVisitDetails;
+import org.gradle.api.specs.Spec;
 import org.gradle.internal.nativeintegration.filesystem.FileSystem;
 
 import java.io.File;
+import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class ReproducibleDirectoryWalker implements DirectoryWalker {
```

Comment:
⭕️ Can we maybe use the Jdk7 walking for the ReproducibleDirectoryWalker as well? I think it would be great to not have two different ways of walking the file system. I think it should be enough to sort the files in one level in `postVisitDirectory` and then notify the listener?

Reply:
Sure, does the `⭕️` mean you're okay with doing it in a different PR?

Reply:
Exactly, the ⭕️ means other PR.